src/test/java/me/karakelley/http/server/HttpServerTest.java

```diff
@@ -464,4 +390,45 @@ private HttpServer configureWithNoDirectory(String port) {
     config.setHandler(new ApplicationFactory().create(args));
     return new HttpServer(config, new ConnectionHandler(), new RequestReaderFactory());
   }
+
+  private void startOnNewThread(HttpServer httpServer) {
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+    executorService.submit(httpServer::start);
+  }
+
+  private void connectClient(HttpServer httpServer, ClientHelper client) {
+    try {
+      client.connectWithTry("127.0.0.1", httpServer);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void connectTwoClients(HttpServer httpServer, ClientHelper client2) {
+    try {
+      client.connectWithTry("127.0.0.1", httpServer);
+      client2.connectWithTry("127.0.0.1", httpServer);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private String basicGetRequest() {
+    return requestBuilder.setMethod("GET").setPath("/").build();
+  }
+
+  private void sleep() {
```

Comment:
I'm confused by this method. Why would an exception be thrown? Also you're incrementing a counter but never looping?